src/Core/Compiler.hpp

```diff
@@ -66,6 +68,14 @@ class Compiler : public QObject
      */
     static bool check(const QString &compileCommand);
 
+    /**
+     * @brief get the output path (executable file path) when compiling C++
+     * @param tmpFilePath the path to the temporary file which is compiled
+     * @param sourceFilePath the path to the original source file, if it's empty, tmpFilePath will be used instead of it
+     * @note the parent directory of the result will be created
+     */
+    static QString cppOutputPath(const QString &tmpFilePath, const QString &sourceFilePath);
```

Comment:
Can you make this function more generic. By returning Exe path, Compiled Java class and "" (filename) for python scripts. Maybe you can change it to `outputPath()`. You will also add a setting for File path for Java solutions class